// Copyright 2022 Cisco Systems, Inc. and its affiliates
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

package grpcMeta;
option go_package = "github.com/cisco-open/flame/pkg/proto/meta";

// Meta info related route
service MetaRoute {
  rpc RegisterMetaInfo(MetaInfo) returns (MetaResponse) {}
  rpc HeartBeat(MetaInfo) returns (MetaResponse) {}
}

message MetaInfo {
  string job_id = 1;
  string ch_name = 2;
  string me = 3;
  string other = 4;
  string endpoint = 5;
}

message MetaResponse {
  enum Status {
    ERROR = 0; // default
    SUCCESS = 1;
  }
  Status status = 1;
  repeated string endpoints = 2;
}
